examples/defaulter-gen/Makefile

```diff
@@ -0,0 +1,15 @@
+TOOL=defaulter-gen
+
+test:
+	@if ! git diff --quiet HEAD; then \
+	    echo "FAIL: git client is not clean"; \
+	    false; \
+	fi
+	@go build -o /tmp/$(TOOL)
+	/tmp/$(TOOL) --logtostderr --v=6 -i ./output_tests/wholepkg -O zz_generated
+	@if ! diff -q ./output_tests/wholepkg/zz_generated.expected ./output_tests/wholepkg/zz_generated.go; then \
```

Comment:
We can change the default-gen Makefile to use --verify-only in a follow-up PR.

For this PR, because the generated code does not compile and break travis, i need to change the suffix of the output file to .expected, so the --verify-only approach doesn't work.

To use --verify-only, we have two options:
* vendor a mock apimachinery folder in the test package to make the code compile.
* hack the .travis to ignore this package.

@lavalamp please let me know how you want me to proceed. Thanks.